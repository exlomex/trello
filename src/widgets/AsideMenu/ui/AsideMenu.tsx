import { classNames } from '@/shared/lib/classNames';
import { FC, useState } from 'react';
import SettingSvg from '@/shared/assets/settings.svg';
import { IconLayout } from '@/shared/layouts/IconLayout';
import { Input } from '@/shared/ui/Input';
import { Button } from '@/shared/ui/Button/Button';
import HideSvg from '@/shared/assets/hide.svg';
import cls from './AsideMenu.module.scss';

interface AsideMenuProps {
    className?: string;
}

export const AsideMenu: FC = ({ className }: AsideMenuProps) => {
    const [collapsed, setCollapsed] = useState(false);

    const toggleCollapsed = () => {
        setCollapsed((prev) => !prev);
        console.log('toggle collapsed to: ', !collapsed);
    };
    return (
        <aside
            className={classNames(
                cls.AsideMenu,
                { [cls.collapsed]: collapsed },
                [className],
            )}
            data-testid={'asideMenu'}
        >
            <div className={classNames(cls.asideUpper, {}, [])}>
                <Input placeholder={'Поиск'} />
                <Button variant={'hideButton'} onClick={toggleCollapsed}>
                    <IconLayout Svg={HideSvg} width={'11px'} height={'19px'} />
                </Button>
            </div>
            <IconLayout
                Svg={SettingSvg}
                clickable={true}
                onClick={() => {}}
                className={cls.settingButton}
            ></IconLayout>
        </aside>
    );
};
