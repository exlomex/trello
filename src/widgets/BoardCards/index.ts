export { BoardCards } from './ui/BoardCards';
