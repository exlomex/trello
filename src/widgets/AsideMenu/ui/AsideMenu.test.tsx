import { render, screen } from '@testing-library/react';
import { AsideMenu } from '@/widgets/AsideMenu';

jest.mock('../../../shared/layouts/IconLayout', () => ({
    IconLayout: jest.fn(),
}));

describe('sidebar', () => {
    test('instanceSidebar', () => {
        render(<AsideMenu />);
        expect(screen.getByTestId('asideMenu')).toBeInTheDocument();
        screen.debug();
    });
});
