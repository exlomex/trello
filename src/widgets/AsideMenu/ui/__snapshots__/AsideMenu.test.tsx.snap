// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sidebar instanceSidebar 1`] = `
<aside
  class="AsideMenu"
  data-testid="asideMenu"
>
  <div
    class="asideUpper"
  >
    <input
      class="Input"
      placeholder="Поиск"
    />
    <button
      class="Button hideButton"
    >
      <div
        class="addonLeft"
      />
      <div
        class="addonLeft"
      />
    </button>
  </div>
</aside>
`;
